File: 2249-count-the-hidden-sequences/2249-count-the-hidden-sequences.java
class Solution {
    public int numberOfArrays (int [] differences, int lower, int upper) {
        long a = 0, max = 0, min = 0;
        for (int i = 0; i < differences.length; i++) {
            a += differences [i];
            max = Math.max (max, a);
            min = Math.min (min, a);
        }
        return (int) Math.max (0, (upper - lower) - (max - min) + 1);
    }
}

File: 2249-count-the-hidden-sequences/2249-count-the-hidden-sequences.py
class Solution:
    def numberOfArrays (self, differences : List [int], lower : int, upper : int) -> int:
        a = maxx = minn = 0
        for i in differences:
            a += i
            maxx = max (maxx, a)
            minn = min (minn, a)
        return max (0, (upper - lower) - (maxx - minn) + 1)